app/models/authentication_token.rb

```diff
@@ -1,3 +1,19 @@
 class AuthenticationToken < ApplicationRecord
+  MAX_TOKEN_DURATION = 1.hour
+
   belongs_to :authenticable, polymorphic: true
+
+  def still_valid?
+    created_at > (Time.zone.now - MAX_TOKEN_DURATION)
+  end
+
+  def self.find_by_hashed_token(authenticable_type:, raw_token:)
+    hashed_token = MagicLinkToken.from_raw(raw_token)
```

Comment:
What's the division of responsibilities between `AuthenticationToken` and `MagicLinkToken` and in practice do we ever have one without the other?

Reply:
As far as i can tell we won't ever have one without the other.